tests/components/demo/test_number.py

```diff
@@ -0,0 +1,73 @@
+"""The tests for the demo number component."""
+
+import pytest
+import voluptuous as vol
+
+from homeassistant.components.number.const import (
+    ATTR_MAX,
+    ATTR_MIN,
+    ATTR_STEP,
+    ATTR_VALUE,
+    DOMAIN,
+    SERVICE_SET_VALUE,
+)
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.setup import async_setup_component
+
+ENTITY_VOLUME = "number.volume"
+
+
+@pytest.fixture(autouse=True)
+async def setup_demo_number(hass):
+    """Initialize setup demo Number entity."""
+    assert await async_setup_component(hass, DOMAIN, {"number": {"platform": "demo"}})
+    await hass.async_block_till_done()
+
+
+def test_setup_params(hass):
+    """Test the initial parameters."""
+    state = hass.states.get(ENTITY_VOLUME)
+    assert state.state == "42.0"
```

Comment:
Is checking that the state is "42.0" robust against floating point formatting and numerical precision? Would it be better to check that `abs(float(state.state) - 42) < 1e-6`, or something like that?

Reply:
In the modern versions of Python, yes. Because the state is a string, python first converts the float number into the string representation. When doing that, it chooses the decimal representation that has the lesser number of fractional digits.
```
Python 3.8.5 (default, Aug 12 2020, 00:00:00)
[GCC 10.2.1 20200723 (Red Hat 10.2.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> print(1/3*3)
1.0
```